Godeps/_workspace/src/github.com/jvatic/asset-matrix-go/generic-asset.go

```diff
@@ -33,6 +38,28 @@ func (a *GenericAsset) Initialize() error {
 	return nil
 }
 
+func (a *GenericAsset) CheckSum() string {
+	if a.checkSum != "" {
+		return a.checkSum
+	}
+	file, err := a.Open()
+	defer file.Close()
+	if err != nil {
+		return ""
+	}
+	s := bufio.NewScanner(file)
+	h := md5.New()
+	for s.Scan() {
+		io.Copy(h, bytes.NewReader(s.Bytes()))
+	}
+	io.WriteString(h, a.p)
+	if err := s.Err(); err != nil {
+		return ""
+	}
+	a.checkSum = fmt.Sprintf("%x", h.Sum(nil))
```

Comment:
I think this could be simplified to:

```go
h := md5.New()
if _, err := io.Copy(h, file); err != nil {
    return ""
}
h.Write([]byte(a.p))
a.checksum = hex.EncodeToString(h.Sum(nil))
```